lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1440,19 +1441,25 @@ def test_progress_queries_paced_courses(self, self_paced, self_paced_enabled):
         """Test that query counts remain the same for self-paced and instructor-paced courses."""
         SelfPacedConfiguration(enabled=self_paced_enabled).save()
         self.setup_course(self_paced=self_paced)
-        with self.assertNumQueries(41), check_mongo_calls(4):
+        with self.assertNumQueries(42), check_mongo_calls(1):
             self._get_progress_page()
 
-    def test_progress_queries(self):
+    @ddt.data(
+        (False, 41, 27),
+        (True, 0, 27)
```

Comment:
So the reason this is failing is due to a problem with how our waffle switch is being overridden. In [this PR](https://github.com/edx/edx-platform/pull/14770), we establish `openedx/core/djangolib/waffle_utils.py`, which basically throws away all of the utility we get from using actual waffle switches (pylint complains `Unused Switch imported from waffle.models` in that file), and instead uses a per-request cached dictionary of namespaced switch names to booleans. Since `_get_progress_page` establishes a brand new request ([source](https://github.com/edx/edx-platform/blob/neem/grade-only-for-engaged/lms/djangoapps/courseware/tests/test_views.py#L1235-L1243)), this cached dictionary does nothing for us, and our switch override is doing nothing. I propose we need to re-think the logic contained in waffle_utils to get this working correctly.

Stack trace, using the added logging in my latest commit. Note that we set the override in one cache, check for it in a totally different empty cache, then clean up the original cache.
```
-------------------- >> begin captured stdout << ---------------------
Read cache
{'switches': {u'block_structure': {}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'grades': {}, u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Set cached value
{'switches': {u'grades': {u'grades.assume_zero_grade_if_absent': True}, u'block_structure': {u'block_structure.storage_backing_for_cache': False, u'block_structure.invalidate_cache_on_publish': False}}}


Read cache
{'switches': {u'grades': {}}}


Read cache
{'switches': {u'grades': {u'grades.assume_zero_grade_if_absent': False}, u'block_structure': {}}}


Read cache
{'switches': {u'grades': {u'grades.assume_zero_grade_if_absent': False}, u'block_structure': {u'block_structure.storage_backing_for_cache': False}}}


Read cache
{'switches': {u'grades': {u'grades.assume_zero_grade_if_absent': False}, u'block_structure': {u'block_structure.storage_backing_for_cache': False}}}


Set cached value
{'switches': {u'grades': {u'grades.assume_zero_grade_if_absent': False}}}



--------------------- >> end captured stdout << ----------------------
```